lib/floe/workflow/states/task.rb

```diff
@@ -51,15 +51,16 @@ def retry!(retrier)
           return if retrier.nil?
 
           # If a different retrier is hit reset the context
-          if !context.key?("retrier") || context["retrier"]["error_equals"] != retrier.error_equals
-            context["retrier"] = {"error_equals" => retrier.error_equals, "retry_count" => 0}
+          if !context["State"].key?("RetryCount") || context["State"]["Retrier"] != retrier.error_equals
```

Comment:
`["State"]["Retrier"]` also isn't in the step functions spec but they also say if a different `ErrorEquals` is matched you have to reset the RetryCount so we need context of _which_ `ErrorEquals` we're in

Reply:
Can we make PRs / suggestions to the spec?  Perhaps that should be there.

Reply:
Well this isn't part of the states-language spec, they just say `This version of the States Language specification does not specify any contents of the Context Object.` [[ref]](https://states-language.net/#context-object) so I was going off of the [[step-functions documentation]](https://docs.aws.amazon.com/step-functions/latest/dg/input-output-contextobject.html) so there isn't a public "spec" as far as I know.

I figured if we provided at least what the aws-step-functions service provides then anything that was written with that in mind would work.